### DISCUSSION TITLE
Get.arguments doesn't work with nested navigation

### DISCUSSION THREAD

**Discussion by euriasb** [jonataslaw/getx#1390] [Q&A] [answered]
Hello, I started using GetX a couple of weeks ago and I'm loving it so far. I'm migrating my current custom navigation to GetX and I'm running into this issue, I'm using nested navigation, specifically I have an iOS style bottom navigation where each bottom navigation item is an independent stack, but I'm doing it with `MaterialApp` and not Cupertino.
For this I'm using an `IndexedStack` with a `Navigator` on each child, this was working perfectly fine before, I was simply using `Navigator.of(context).pushNamed('/route', arguments: {'arg': arg})` and the route would be pushed to the current `Navigator`, I switched it to use GetX, made changes to each tab's `Navigator` and now I'm able to do `Get.toNamed('/route', id: activeTabId)`, the problem is that if I do `Get.arguments` I'm not getting the arguments passed in `Get.toNamed`, after debugging I found out as I suspected that Get thinks that the current route is the "root" route, i.e. the one containing the bottom navigation, not the one nested inside of it, so my question is if this is working as intended or if it is a bug and how can I get the arguments passed here.

Thanks.

**Comment by euriasb** [comment, marked as answer]
Not surprisingly, as soon as I asked I found the answer, if anyone has the same issue, you need to add an observer to your Navigator, i.e.

```
Navigator(
  ...
  observers: [GetObserver(null, Get.routing)],
);
```

**Comment by dirablue** [comment]
@euriasb 
Thank you! it works.
